File: 8_pascals_triangle/pascals_triangle.cpp
#include <iostream>
#include <ctime>
using namespace std;

int main() {
    cout << "Enter the number of Pascal Triangle rows--> " << endl;
    int rows;
    cin >> rows;
    time_t start_time = time(NULL); //Start the timer
    static int triangle[2000][2000] = {0}; //Fill with all 0's, don't print the 0's later

    for(int i = 0; i < rows + 1; i++) {
        triangle[i][0] = 1; //First element 1
        triangle[i][i] = 1; //Last element 1

        for(int j = 1; j < i; j++) { 
            triangle[i][j] = triangle[i-1][j-1] + triangle[i-1][j]; //Add number above to the left and just above
        }
    }

    for(int k = 0; k < rows; k++) {
        for(int n = 0; n < rows; n++) {
            if (triangle[k][n] != 0) {
            cout << triangle[k][n] << " ";
            }
        }
        cout << "\n";
    }
    time_t end_time = time(NULL);
    cout << "\nTime taken: " << end_time - start_time << " seconds" << endl;
    return 0;
}

File: 8_pascals_triangle/pascals_triangle.py
import time

def main():
    rows = int(input("Enter the number of Pascal Triangle rows--> "))

    start_time = time.time()
    #Initialize a 2000x2000 list with all 0s
    triangle = [[0 for _ in range(2000)] for _ in range(2000)]

    for i in range(rows + 1):
        triangle[i][0] = 1  #First element 1
        triangle[i][i] = 1  #Last element 1

        for j in range(1, i):
            triangle[i][j] = triangle[i-1][j-1] + triangle[i-1][j]  #Add elements above to the left and directly above

    #Print Pascal's Triangle
    for k in range(rows):
        for n in range(rows):
            if triangle[k][n] != 0:
                print(triangle[k][n], end=" ")
        print()
    end_time = time.time()
    execution_time = end_time - start_time
    print(execution_time)

if __name__ == "__main__":
    main()